hub/tests/dataset_fixtures.py

```diff
@@ -47,7 +47,9 @@ def generate_s3_ds():
 
 @pytest.fixture
 def hub_cloud_ds(hub_cloud_ds_generator):
-    return hub_cloud_ds_generator()
+    hub_cloud_dataset = hub_cloud_ds_generator()
+    yield hub_cloud_dataset
+    hub_cloud_dataset.delete()
```

Comment:
this is not a good idea, the generator is used to test persistence. this means that all tests checking persistence with hub cloud datasets **will** fail

Reply:
you should update the hub cloud dataset generator fixture